Android/app/src/main/java/com/example/brewhaha_android/Controllers/HomeActivity.kt

```diff
@@ -1,48 +1,79 @@
 package com.example.brewhaha_android.Controllers
 
+import android.app.ProgressDialog
 import android.content.Intent
 import android.os.Bundle
 import android.util.Log
-import android.widget.TextView
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.view.View
+import android.widget.ImageView
 import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
 import com.example.brewhaha_android.Api.BackendConnection
+import com.example.brewhaha_android.Models.AuthToken
+import com.example.brewhaha_android.Models.Brewery
 import com.example.brewhaha_android.Models.LogoutUser
 import com.example.brewhaha_android.R
 import com.google.android.material.button.MaterialButton
+import com.google.android.material.textview.MaterialTextView
 import kotlinx.android.synthetic.main.activity_home.*
 import org.jetbrains.anko.doAsync
 import org.jetbrains.anko.toast
 import org.jetbrains.anko.uiThread
 
 class HomeActivity(private val api: BackendConnection = BackendConnection()) : AppCompatActivity() {
     var _logout_button : MaterialButton? = null
-    var _view_brewery_button : MaterialButton? = null
+    var _filter_button : MaterialButton? = null
+    var tokenBundle: Bundle? = null
+    lateinit var breweryList: List<Brewery>
+    private lateinit var linearLayoutManager: LinearLayoutManager
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_home)
 
+        // Logout stuff
         _logout_button = findViewById<MaterialButton>(R.id.logoutButton)
-        _view_brewery_button = findViewById(R.id.viewBreweries)
-
-        var tokenBundle = intent.getBundleExtra("bundle")
-        var token = tokenBundle["token"] as String
-        var refreshToken = tokenBundle["refreshToken"] as String
-        var userId = tokenBundle["id"] as String
-
+        tokenBundle = intent.getBundleExtra("bundle")
+        val userId = tokenBundle!!["id"] as String
         _logout_button!!.setOnClickListener{
             logout(userId)
         }
 
-        _view_brewery_button!!.setOnClickListener{
-            viewBreweries(tokenBundle)
+        _filter_button = findViewById<MaterialButton>(R.id.filterButton)
+        _filter_button!!.setOnClickListener {
+        // TODO: make filter sheet
+        }
+
+        val progressDialog = ProgressDialog(this)
+        progressDialog.isIndeterminate = true
+        progressDialog.setMessage("Finding Breweries...")
+        progressDialog.show()
+
+        // Get brewery stuff
+        linearLayoutManager = LinearLayoutManager(this)
+        Log.d("Home", "About to get breweries")
+        Log.d("Home", "GetBreweries = ")
+        getBreweries()
+        while (!(::breweryList.isInitialized)) {
```

Comment:
Instead of doing this can you do it all inside of an async call.
What I mean by that is in the getBreweries() function, have a async call to the backend and then when you get a success response in the uiThread{} update the listView and the container view. For now just give it some dummy list or make it empty when it first renders the page.